src/MainWindow.vala

```diff
@@ -273,18 +273,28 @@ public class Maps.MainWindow : Adw.ApplicationWindow {
             answer_count = 10
         };
 
-        loc_store.remove_all ();
-
         var places = new List<Geocode.Place> ();
         try {
             places = yield forward.search_async (search_cancellable);
         } catch (Error error) {
             warning (error.message);
+            loc_store.remove_all ();
             return;
         }
 
+        // Remove any old results that aren't in the new set
+        for (int i = 0; i < loc_store.n_items; i++) {
+            if (places.find ((Geocode.Place) loc_store.get_item (i)) == null) {
```

Comment:
I guess `GLib.List.find()` is pointer comparison? So I'm not sure if this works as you expect.

Reply:
Yeah I wasn't sure about it, but from testing it seems to work. I had it print number added and number removed and most of the time they were not equal amounts and there was matches to be removed :woman_shrugging:

Reply:
> Maybe find with equal func would be better?

@danirabbit Yes, you should use `GLib.List.find_custom()` instead if you really want to do this, but honestly I'm not sure if removing/adding results is better than clearing the liststore……I know this is an application code but we call `find()` in the list, which I don't think light call, twice in this code.

Reply:
Okay gonna just close. This might be trying to over-optimize. We can always revisit it later if there's a real performance issue to look at